axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceToolService.java

```diff
@@ -46,18 +48,38 @@ public static LocalDate getDueDate(Invoice invoice) throws AxelorException {
     if (CollectionUtils.isEmpty(invoice.getInvoiceTermList())) {
       return invoiceDate;
     }
-    LocalDate dueDate = null;
-    for (InvoiceTerm invoiceTerm : invoice.getInvoiceTermList()) {
-      if (!invoiceTerm.getIsPaid()
-          && (dueDate == null || invoiceTerm.getDueDate().isBefore(dueDate))) {
-        dueDate = invoiceTerm.getDueDate();
-      }
+    if (invoice.getInvoiceTermList().size() == 1) {
+      return invoice.getInvoiceTermList().get(0).getDueDate();
     }
+    return invoice.getInvoiceTermList().stream()
+        .map(invoiceTerm -> invoiceTerm.getDueDate())
+        .max(Comparator.comparing(LocalDate::toEpochDay))
+        .get();
```

Comment:
get() will throw an exception if nothing is found, and not return `null`, please fix it